examples/regression/1-quickstart/plot_regression_housing_dataset.py

```diff
@@ -0,0 +1,426 @@
+# We notice more adaptability of the prediction intervals for the
+# conformalized quantile regression while the other methods have fixed
+# interval width.
+
+
+def get_bins(
+    want: str,
+    y_test: Dict,
+    y_pred: Dict,
+    lower_bound: Dict,
+    upper_bound: Dict,
+    STRATEGIES: Dict,
+    bins: ArrayLike
+) -> pd.DataFrame:
+    """
+    Splits the data into different sections according to the bins selected.
+    Computes the coverage and width of that split.
+
+    Parameters
+    ----------
+    want : str
+        _description_
+    y_test : Dict
+        _description_
+    y_pred : Dict
+        _description_
+    lower_bound : Dict
+        _description_
+    upper_bound : Dict
+        _description_
+    STRATEGIES : Dict
+        _description_
+    bins : ArrayLike
+        _description_
+
+    Returns
+    -------
+    pd.DataFrame
+        _description_
+    """
+    cuts = []
+    cuts_ = pd.qcut(y_test["naive"], bins).unique()[:-1]
+    for item in cuts_:
+        cuts.append(item.left)
+    cuts.append(cuts_[-1].right)
+    cuts.append(np.max(y_test["naive"])+1)
+    recap = {}  # type: ignore
+    for i in range(len(cuts)-1):
+        cut1, cut2 = cuts[i], cuts[i+1]
+        name = f"[{np.round(cut1, 0)}, {np.round(cut2, 0)}]"
+        recap[name] = []
+        for strategy in STRATEGIES:
+            indices = np.where(
+                (y_test[strategy] > cut1)*(y_test[strategy] <= cut2)
+                )
+            y_test_trunc = np.take(y_test[strategy], indices)
+            y_low_ = np.take(lower_bound[strategy], indices)
+            y_high_ = np.take(upper_bound[strategy], indices)
+            if want == "coverage":
+                recap[name].append(regression_coverage_score(
+                    y_test_trunc[0],
+                    y_low_[0],
+                    y_high_[0]))
+            elif want == "width":
+                recap[name].append(
+                    regression_mean_width_score(y_low_[0], y_high_[0])
+                    )
+    recap_df = pd.DataFrame(recap, index=STRATEGIES)
+    return recap_df
+
+
+bins = list(np.arange(0, 1, 0.1))
+binned_data = get_bins(
+    "coverage",
+    y_test_sorted,
+    y_pred_sorted,
+    lower_bound,
+    upper_bound,
+    STRATEGIES,
+    bins
+    )
+
+
+##############################################################################
+# To confirm this insights, we will now observe what happens when we plot
+# the conditional coverage and interval width on these intervals.
+
+
+binned_data.T.plot.bar(figsize=(12, 4))
+plt.axhline(0.80, ls="--", color="k")
+plt.ylabel("Conditional coverage")
+plt.xlabel("Binned house prices")
+plt.xticks(rotation=345)
+plt.ylim(0.3, 1.0)
+plt.legend(loc=[1, 0])
+
+
+##############################################################################
+# What we observe from these results is that none of the methods seems to
+# have conditional coverage. It is however suprising to see that the
```

Comment:
have conditional coverage at the target :math:`1 - \alpha` level.

Reply:
done